config/validated_config_db_connector.py

```diff
@@ -0,0 +1,47 @@
+import jsonpatch
+from jsonpointer import JsonPointer
+
+from generic_config_updater.generic_updater import GenericUpdater, ConfigFormat
+from generic_config_updater.gu_common import EmptyTableError
+from swsscommon.swsscommon import ConfigDBConnector
+
+def ValidatedConfigDBConnector(config_db_connector, fallback_adhoc_validation):
+    if not fallback_adhoc_validation:
+        config_db_connector.set_entry = validated_set_entry
+    return config_db_connector
+
+def make_path_value_jsonpatch_compatible(table, key, value):
+    if type(key) == tuple:
+        path = JsonPointer.from_parts([table, '|'.join(key)]).path
+    else:
+        path = JsonPointer.from_parts([table, key]).path
+    if value == {"NULL" : "NULL"}:
+        value = {}
+    return path, value
+
+def validated_set_entry(table, key, value):
+    if value is not None:
+        op = "add"
+    else:
+        op = "remove"
+    path, value = make_path_value_jsonpatch_compatible(table, key, value) 
+
+    gcu_json_input = []
+    gcu_json = {"op": "{}".format(op),
+                "path": "{}".format(path)}
+    if op == "add":
+        gcu_json["value"] = value
+
+    gcu_json_input.append(gcu_json)
+    gcu_patch = jsonpatch.JsonPatch(gcu_json_input)
+    format = ConfigFormat.CONFIGDB.name
+    config_format = ConfigFormat[format.upper()]
+
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    try:
+        GenericUpdater().apply_patch(patch=gcu_patch, config_format=config_format, verbose=False, dry_run=False, ignore_non_yang_tables=False, ignore_paths=None)
+    except EmptyTableError:
```

Comment:
>EmptyTableError [](http://example.com/codeflow?start=11&length=15)

I understand GCU could not delete the last vlan entry in vlan table, because it will lead to an empty vlan table. And GCU is okay to delete the whole vlan table in this case.

Suggest catch the exception, if the original patch is delete one entry, we change the patch to delete the whole table and retry `apply_patch`.

Reply:
Or you can decorate `delete_table`, and call it instead.

Reply:
Or
```
gcu_json_input.append(gcu_json)
gcu_json_input.append({
    "op": "remove",
    "path": table
})
```

Reply:
I decorated `delete_table` and changed exception handling to call `validated_delete_table`